src/replication/adapter/quayio/image_registry.go

```diff
@@ -0,0 +1,41 @@
+package quayio
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net/http"
+
+	"github.com/docker/distribution"
+	"github.com/docker/distribution/manifest/schema1"
+	registry_pkg "github.com/goharbor/harbor/src/common/utils/registry"
+)
+
+// PullManifest get manifest details from Quay.io
+func (a *adapter) PullManifest(repository, reference string, accepttedMediaTypes []string) (distribution.Manifest, string, error) {
```

Comment:
Is there any special for the PullManifest with the implementation of docker registry(native) adapter?

Reply:
At first, use native adapter pull manifest will cause "missing signatures key" error, but it can be worked with native implement, i will remove the function later.